lib/Event/AbstractManagerEvent.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\DocumentManager\Event;
+
+use Sulu\Component\DocumentManager\DocumentManagerInterface;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Abstract class for events which require the DocumentManagerInterface - i.e.
+ * all events for which subscribers require access to a document manager or any
+ * of its dependencies.
+ */
+abstract class AbstractManagerEvent extends AbstractEvent
+{
+    private $manager;
+
+    /**
+     * @param DocumentManagerInterface $manager
+     */
+    public function __construct(DocumentManagerInterface $manager)
+    {
+        $this->manager = $manager;
+    }
+
+    public function getManager()
+    {
+        if (null === $this->manager) {
+            throw new \RuntimeException(
+                'No DocumentManagerInterface instance has been set on this event, maybe this class has overridden the constructor and forgotten about it?'
```

Comment:
It is overridden by other classes which override the __constructor.
Better safe than sorry.

On Thu, Apr 14, 2016 at 01:11:26AM -0700, Daniel Rotter wrote:
>    In [1]lib/Event/AbstractManagerEvent.php:
> 
>  > +{
>  > +    private $manager;
>  > +
>  > +    /**
>  > +     * @param DocumentManagerInterface $manager
>  > +     */
>  > +    public function __construct(DocumentManagerInterface $manager)
>  > +    {
>  > +        $this->manager = $manager;
>  > +    }
>  > +
>  > +    public function getManager()
>  > +    {
>  > +        if (null === $this->manager) {
>  > +            throw new \RuntimeException(
>  > +                'No DocumentManagerInterface instance has been set on this event, maybe this class has overridden the constructor and forgotten about it?'
> 
>    And is this check necessary at all? The constructor always sets the
>    manager, and because of the typehint it's not allowed to be null.
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly or [2]view it on GitHub
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu/sulu-document-manager/pull/66#discussion_r59677296
>    2. https://github.com/sulu/sulu-document-manager/pull/66/files/6e7fab9301c97bff800cfff05d913f78e432ded1#r59677296
>    3. https://github.com/sulu/sulu-document-manager/pull/66/files/6e7fab9301c97bff800cfff05d913f78e432ded1#r59677296